src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs

```diff
@@ -70,6 +70,29 @@ public Response<List<PlaceholderNameAndValue>> Find(string path, string query, s
 
                     counterForTemplate = endOfPlaceholder;
                 }
+                else if (TemplateDoesNotEndInForwardSlash(pathTemplate) && IsLastPlaceholder(path, counterForPath, pathTemplate))
+                {
```

Comment:
The code for this **if**-block was copied from lines 26-46.
Based on the function of code snippet, why not to define private helper-method to be reused in both places?